diff -up SQLAlchemy-0.5.5/test/aaa_profiling/test_pool.py.profile-for-2.6.6 SQLAlchemy-0.5.5/test/aaa_profiling/test_pool.py
--- SQLAlchemy-0.5.5/test/aaa_profiling/test_pool.py.profile-for-2.6.6	2012-01-28 09:49:14.997265566 -0500
+++ SQLAlchemy-0.5.5/test/aaa_profiling/test_pool.py	2012-01-28 09:50:09.979153364 -0500
@@ -15,7 +15,8 @@ class QueuePoolTest(TestBase, AssertsExe
                          use_threadlocal=True)
 
 
-    @profiling.function_call_count(54, {'2.4': 38})
+    @profiling.function_call_count(54, {'2.4': 38,
+                                        '2.6.6': 49})
     def test_first_connect(self):
         conn = pool.connect()
 
@@ -23,7 +24,8 @@ class QueuePoolTest(TestBase, AssertsExe
         conn = pool.connect()
         conn.close()
 
-        @profiling.function_call_count(31, {'2.4': 21})
+        @profiling.function_call_count(31, {'2.4': 21,
+                                            '2.6.6': 28})
         def go():
             conn2 = pool.connect()
             return conn2
